### PR TITLE
Add tutorial to use a Pytorch model

### PR DESCRIPTION
**Pull request by doreenfan**

### CODE REVIEW

File: ML/PYTORCH/Source/main.cpp [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+            int ii = i - bx_lo[0];
+            int jj = j - bx_lo[1];
+            int index = jj*nbox[0] + ii;
+#if AMREX_SPACEDIM == 3
+            int kk = k - bx_lo[2];
+            index += kk*nbox[0]*nbox[1];
+#endif
+            // array order is row-based [index][comp]
+            auxPtr[index*Nc_in + 0] = phi_input(i, j, k, 0);
+        });
```
Comment (ajnonaka) [inline ML/PYTORCH/Source/main.cpp:218]:
```suggestion
        amrex::ParallelFor(bx, Nc_in, [=] AMREX_GPU_DEVICE(int i, int j, int k, int n) noexcept
        {
            int ii = i - bx_lo[0];
            int jj = j - bx_lo[1];
            int index = jj*nbox[0] + ii;
#if AMREX_SPACEDIM == 3
            int kk = k - bx_lo[2];
            index += kk*nbox[0]*nbox[1];
#endif
            // array order is row-based [index][comp]
            auxPtr[index*Nc_in + n] = phi_input(i, j, k, n);
        });
```

File: ML/PYTORCH/Source/main.cpp [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+    BL_PROFILE_VAR("InitData",InitData);
+
+    // loop over boxes
+    for (MFIter mfi(phi_in); mfi.isValid(); ++mfi)
+    {
+        const Box& bx = mfi.validbox();
+
+        const Array4<Real>& phi_input = phi_in.array(mfi);
+
+        // set phi = random(0, dt)
```
Comment (ajnonaka) [inline ML/PYTORCH/Source/main.cpp:130]:
```suggestion
        // set phi
```

File: ML/PYTORCH/Source/main.cpp [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+        });
+    }
+
+    BL_PROFILE_VAR_STOP(InitData);
+
+    BL_PROFILE_VAR("WriteInitPlot",WriteInitPlot);
+
+    // Write a plotfile of the initial data
+    const std::string& pltfile = amrex::Concatenate("plt",0,5);
+    WriteSingleLevelPlotfile(pltfile, phi_in, {"dt"}, geom, 0.0, 0);
```
Comment (ajnonaka) [inline ML/PYTORCH/Source/main.cpp:143]:
```suggestion
    WriteSingleLevelPlotfile(pltfile, phi_in, {"x0"}, geom, 0.0, 0);
```

File: ML/PYTORCH/Source/main.cpp [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+
+    // size of each box (or grid)
+    int max_grid_size;
+
+    // pytorch model file
+    std::string model_filename;
+
+    // largest time step
+    Real dt;
+
```
Comment (ajnonaka) [inline ML/PYTORCH/Source/main.cpp:43]:
```suggestion
```

File: ML/PYTORCH/Source/main.cpp [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+        pp.get("n_cell",n_cell);
+
+        // The domain is broken into boxes of size max_grid_size
+        pp.get("max_grid_size",max_grid_size);
+
+        // Read in model file name
+        pp.query("model_file", model_filename);
+
+        // time step
+        pp.get("dt", dt);
```
Comment (ajnonaka) [inline ML/PYTORCH/Source/main.cpp:62]:
```suggestion
```

File: ML/PYTORCH/Source/main.cpp [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+    BL_PROFILE_VAR_STOP(Eval);
+
+    BL_PROFILE_VAR("Post",Post);
+
+    // Tell the I/O Processor to write out that we're done
+    amrex::Print() << "Finish evaluating model.\n";
+
+    // Write a plotfile of the current data
+    const std::string& pltfile2 = amrex::Concatenate("plt",1,5);
+    WriteSingleLevelPlotfile(pltfile2, phi_out, {"y0", "y1"}, geom, dt, 1);
```
Comment (ajnonaka) [inline ML/PYTORCH/Source/main.cpp:263]:
```suggestion
    WriteSingleLevelPlotfile(pltfile2, phi_out, {"y0", "y1"}, geom, 1.0, 1);
```

File: ML/PYTORCH/Exec/inputs [hunk 7]
```diff
@@ -0,0 +1,4 @@
+n_cell = 64
+max_grid_size = 64
+model_file = "model.pt"
+dt = 1.0e1
```
Comment (ajnonaka) [inline ML/PYTORCH/Exec/inputs:4]:
```suggestion
```

File: ML/PYTORCH/README.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+```shell
+# CPU only
+wget https://download.pytorch.org/libtorch/cpu/libtorch-cxx11-abi-shared-with-deps-1.9.0%2Bcpu.zip
+unzip libtorch-cxx11-abi-shared-with-deps-1.9.0+cpu.zip
+mv libtorch libtorch_cpu
+
+# CUDA 11.1
+wget https://download.pytorch.org/libtorch/cu111/libtorch-cxx11-abi-shared-with-deps-1.9.0%2Bcu111.zip
+unzip libtorch-cxx11-abi-shared-with-deps-1.9.0%2Bcu111.zip
```
Comment (ajnonaka) [inline ML/PYTORCH/README.md:14]:
```suggestion
unzip libtorch-cxx11-abi-shared-with-deps-1.9.0+cu111.zip
```

File: ML/PYTORCH/README.md [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+mv libtorch libtorch_cuda
+
+# CUDA 10.2
+wget https://download.pytorch.org/libtorch/cu111/libtorch-cxx11-abi-shared-with-deps-1.9.0%2Bcu102.zip
```
Comment (ajnonaka) [inline ML/PYTORCH/README.md:18]: This doesn't work for me - says "ERROR 405: Forbidden"

File: ML/PYTORCH/README.md (hunk 8 above)
Comment (ajnonaka) [reply ML/PYTORCH/README.md:14]: ^ this was the actual file that it downloaded.  Does it do the same for you?

Comment (doreenfan) [reply ML/PYTORCH/README.md:14]: Ah that was a typo

File: ML/PYTORCH/README.md (hunk 9 above)
Comment (doreenfan) [reply ML/PYTORCH/README.md:18]: oops, that's another typo

**Pull request merged**